yarl/_path.py

```diff
@@ -39,3 +40,31 @@ def normalize_path(path: str) -> str:
 
     segments = path.split("/")
     return prefix + "/".join(normalize_path_segments(segments))
+
+
+def calculate_relative_path(target: str, base: str) -> str:
+    """Return the relative path between two other paths.
+
+    If the operation is not possible, raise ValueError.
+    """
+
+    target = target or "/"
+    base = base or "/"
+
+    target_path = PurePosixPath(target)
+    base_path = PurePosixPath(base)
+
+    if not base.endswith("/"):
+        base_path = base_path.parent
+
+    for step, path in enumerate([base_path] + list(base_path.parents)):
```

Comment:
Can we replace the creation of two additional in-memory lists with `itertools.chain`?

Reply:
It looks like this can be a simple unpack. Addressed in https://github.com/aio-libs/yarl/pull/1379

Reply:
chain was a little bit better... performance is still quadratic in most cases though :(

https://github.com/aio-libs/yarl/pull/1382